module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -202,9 +202,10 @@ kv_alloc(spl_kmem_cache_t *skc, int size, int flags)
 	if (skc->skc_flags & KMC_KMEM) {
 		ASSERT(ISP2(size));
 		ptr = (void *)__get_free_pages(lflags, get_order(size));
-	} else if (skc->skc_flags & KMC_KVMEM) {
-		ptr = spl_kvmalloc(size, lflags);
-	} else {
+		ASSERT(IS_P2ALIGNED(ptr, PAGE_SIZE));
+	} else if ((((lflags & GFP_KERNEL) == GFP_KERNEL) ||
+		    (skc->skc_flags & KMC_KVMEM)) &&
+		   (skc->skc_obj_align > SPL_KMEM_CACHE_ALIGN)) {
```

Comment:
Shouldn't this check be `skc->skc_obj_align <= SPL_KMEM_CACHE_ALIGN`?

Reply:
meh. copy-pasta. thanks